### PR TITLE
[Merged by Bors] - Delete orphaned resources

### LINKED ISSUE #177: Handle stale information/clean up stale resources
Currently our operators will not act on removed information from the CR in some/most/all cases.

One example:
HBase operator has three roles (master, regionServer, restServer). If I create a HBase server CR with a restServer component and then remove it later (entirely, not setting replicas to 0) our operator will not clean up the STS that belongs to this role.

## Proposed solution

Use the `ClusterResource` struct from operator-rs to manage Kubernetes resources belonging to a Cluster object. An example of its usage can be found in the Superset Operator: https://github.com/stackabletech/superset-operator/blob/main/rust/operator-binary/src/superset_controller.rs#L241

## Acceptance criteria

- [x] This is done when all stale Kubernetes resources are cleaned up. A resource becomes stale when it's not part of the current cluster definition anymore.
- [x] `TrinoCatalog` resources are not deleted because the operator cannot know if they are stale or not.
- [x] There is at least one test.
- [ ] Documentation on operator implementation is updated with information regarding handling of stale resources.
- [x] Upgrade to the latest version of` operator-rs` (`0.25` at the moment).

NOTE: This is part of an epic (https://github.com/stackabletech/issues/issues/203) and _might_ not apply to this operator. If that is the case please comment on this issue and just close it. This issue was created as part of a special bulk creation operation.

### PR DESCRIPTION
**Pull request by siegfriedweber**
# Description

Delete orphaned resources

Closes #177 

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/operator-binary/src/controller.rs [hunk 1]
```diff
@@ -931,7 +942,7 @@ async fn create_shared_internal_secret(trino: &TrinoCluster, client: &Client) ->
         .is_none()
     {
         client
-            .apply_patch(FIELD_MANAGER_SCOPE, &secret, &secret)
+            .apply_patch(CONTROLLER_NAME, &secret, &secret)
```
Comment (sbernauer) [inline rust/operator-binary/src/controller.rs:945]: I would think that it makes sense to include the shared secret as well. I does not need to stay persistent between Trino Clusters as far as i know. CC @maltesander

Comment (maltesander) [reply rust/operator-binary/src/controller.rs:945]: Yeah each cluster creates a shared secret for internal communication. So they are independent.

Comment (sbernauer) [reply rust/operator-binary/src/controller.rs:945]: Nevermind, they have a owner reference

**Review by sbernauer — APPROVED** [review]

**Comment by siegfriedweber** [comment]
bors merge

**Pull request closed without merging**